airsonic-main/src/main/java/org/airsonic/player/service/search/IndexManager.java

```diff
@@ -62,14 +65,40 @@
 
     private static final Logger LOG = LoggerFactory.getLogger(IndexManager.class);
 
-    // XXX 3.x -> 8.x : #1166 Holds a tentative literal. It needs to be resolved by the release.
-    private static final String INDEX_ROOT_DIR_NAME = "lucene2";
+    /**
+     * Schema version of Airsonic index.
+     * It may be incremented in the following cases:
+     * 
+     *  - Incompatible update case in Lucene index implementation
+     *  - When schema definition is changed due to modification of AnalyzerFactory,
+     *    DocumentFactory or the class that they use.
+     *
+     */
+    private static final int INDEX_VERSION = 16;
+
+    /**
+     * Literal name of index top directory.
+     */
+    private static final String INDEX_ROOT_DIR_NAME = "index";
```

Comment:
Thanks for the directory name change.  But could I perhaps suggest using "search" rather than index?  Index could mean lots of things other than lucene.

Reply:
It might also be noted in a comment here that the directory name should be changed (like with a version suffix) whenever a change that causes index incompatibility is made.  In fact, maybe it should be made `search3` now so that is clear since the last one was `lucene2`?

Reply:
Indexes seem to be appropriate from past sources, but let me ask muff1nman to decide.

Reply:
That?
I think it was to be a literal, but my English ability may have a misunderstanding.
Ask muff1nman for confirmation.

Reply:
`index` is fine, but like @eharris said I think it'd be good to have a version suffix. So `index3`?

Reply:
Or using the existing `INDEX_VERSION` constant would be fine too. So in that case the directory would be named `index16`